### PR TITLE
[Bugfix] update user policy

### PR DESCRIPTION
**Pull request by tristan-orourke**
## 👋 Introduction

Updates the User Policy permission to correctly protect roleAssignment and sub.

## 🕵️ Details

This is the bare-minimum quick fix. I'd like to change the updateUserAsAdmin mutation a bit further, to make these kinds of bugs easier to avoid in the future.

I added an one more permission, `update-any-userSub`, to reflect the fact that admins can update MORE fields on the user object than users can touch themselves. Also fixed the field name used to check for role assignment.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. set `AUTH_DEFAULT_USER='applicant@test.com'` in api/.env 
2. Go to http://localhost:8000/graphiql
3. As a non-admin user, try to edit your own sub field. You can find <applicant-user-id> by running the `me` query. The following mutation should fail:
```
mutation UpdateUserAsAdmin {
  updateUserAsAdmin(    
    id: "<applicant-user-id>" 
    user: {sub: "applicant@test.com_edited"} 
  ) {
    id
    sub
    email
  }
}
```
4. As a non-admin user, try to edit your own sub field. The role id can be found by looking in the database, or by running the `roles` as an admin. The following mutation should fail:
```
mutation UpdateUserAsAdmin {
  updateUserAsAdmin(
    user: {sub: "applicant@test.com", roleAssignmentsInput: {attach: {roles: "<platform-admin-role-id>"}}}
    id: "29ffd391-aa4c-4386-8f68-e704f904fb92"
  ) {
    id
    sub
    email
    roleAssignments {
      id
      role {
        name
      }
      team {
        name
      }
    }
  }
}
```

## 🚚 Deployment

Run `php artisan db:seed --class=RolePermissionSeeder`

### CODE REVIEW

File: api/config/rolepermission.php
Comment (vd1992) [inline api/config/rolepermission.php:1]: You've added a new resource and a new permission, but it isn't seeded for platform_admin

Comment (tristan-orourke) [reply api/config/rolepermission.php:1]: 🤦‍♂️ Thanks, fixed

Comment (vd1992) [reply api/config/rolepermission.php:1]: For completeness sake, please add it to `RolePermissionTest.php` too 😛

Comment (vd1992) [reply api/config/rolepermission.php:1]: Oh, and it appears you don't have Pint running on save to catch style/formatting (for linting)

**Comment by vd1992** [comment]
Before

image

This branch

image

File: api/app/Policies/UserPolicy.php [hunk 1]
```diff
@@ -63,17 +63,21 @@ public function create(User $user)
...
+         * If a user is assigning a role or updating 'sub', check for extra permissions and fail early
          */
-        if ($injected && isset($injected['roles'])) {
+        if ($injected && isset($injected['roleAssignmentsInput'])) {
             if (! $user->isAbleTo('assign-any-role')) {
                 return false;
             }
         }
+        if ($injected && isset($injected['sub'])) {
+            if (! $user->isAbleTo('assign-any-userSub')) {
```
Comment (vd1992) [inline api/app/Policies/UserPolicy.php:76]: The new permission added is "update-any-userSub" not using "assign"

**Review by vd1992 — APPROVED** [review]
Good! Just the tiny Pint thing left, the trailing comma.
Can run it locally as well with: 

`docker-compose exec -w /home/site/wwwroot/api webserver sh -c "vendor/bin/pint"`

**Comment by tristan-orourke** [comment]
> Good! Just the tiny Pint thing left, the trailing comma. Can run it locally as well with:
> 
> `docker-compose exec -w /home/site/wwwroot/api webserver sh -c "vendor/bin/pint"`

Thanks! I have a plugin installed for Pint, but it seems like it wasn't running automatically.

**Pull request merged**